src/wazuh_modules/inventory_harvester/src/systemInventory/elements/portElement.hpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Wazuh Vulnerability scanner - Scan Orchestrator
+ * Copyright (C) 2015, Wazuh Inc.
+ * March 20, 2025.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _PORT_ELEMENT_HPP
+#define _PORT_ELEMENT_HPP
+
+#include "../../wcsModel/data.hpp"
+#include "../../wcsModel/inventoryPortHarvester.hpp"
+#include "../../wcsModel/noData.hpp"
+#include <stdexcept>
+
+template<typename TContext>
+class PortElement final
+{
+public:
+    // LCOV_EXCL_START
+    /**
+     * @brief Class destructor.
+     *
+     */
+    ~PortElement() = default;
+    // LCOV_EXCL_STOP
+
+    static DataHarvester<InventoryPortHarvester> build(TContext* data)
+    {
+        auto agentId = data->agentId();
+        if (agentId.empty())
+        {
+            throw std::runtime_error("Agent ID is empty, cannot upsert system element.");
```

Comment:
This message is the same for all the elements, I don't know if that can cause confusion in a hypotetical debugging session, but I changed `system` by `element_name` (hotfix in my case)

Reply:
You are right, I got confused with the `system` word. Added.